### DISCUSSION TITLE
What is UnifiedJedis for?

### DISCUSSION THREAD

**Discussion by zeekling** [xetorthio/jedis#2734] [Q&A] [answered]
What is UnifiedJedis for? I don't quite understand the reason for adding this.

I want to fix the following test error
![image](https://user-images.githubusercontent.com/17929800/145699526-833f5ef8-524e-470a-995b-5ad9ca14253a.png)

but don't know What is UnifiedJedis for

**Comment by sazzad16** [comment, marked as answer]
@zeekling 

1. UnifiedJedis could be anything depending on connection provider or executor. In the main codebase, this is the base for JedisCluster, JedisPooled, JedisSharding.
2. As the name of test class suggests, those are base for test classes which should be avoided during testing.
